veps/sig-compute/vmtemplate-proposal.md

```diff
@@ -0,0 +1,733 @@
+create a new `VirtualMachineTemplate` from the results. This new template can
+then be used to create actual `VirtualMachines` from the imported template.
+
+Similar to the import mechanism, the entry point to the export mechanism should
+be a new `VirtualMachineTemplateExportRequest` CRD containing the required
+fields to specify either an existing `VirtualMachineTemplate` to be exported or
+an existing `VirtualMachine` to be exported directly. Once the exported template
+is ready to be downloaded, the controller should make the URL to download the
+template available in the status of the new CRD.
+
+For both import and export, `virtctl` should be extended to enable the feature
+to be used easily via the CLI.
+
+## API Examples
+
+For API examples, see the code blocks in the `template.kubevirt.io/v1alpha1`
+design.
+
+## Alternatives
+
+### Develop feature inside the core KubeVirt repository
+
+The current design proposes to develop this feature outside the KubeVirt core
+repository. An alternative would be to develop the feature within the KubeVirt
+core repository. However, this would lead to a slowdown in development as it
+would further increase the burden on reviewers in the core repository, would
+require all KubeVirt e2e tests to be run, most of which have nothing to do with
+the feature, and it would further increase the already substantial codebase.
+
+### Use existing k8s workload template solutions such as Helm or Kustomize
+
+As discussed within the Motivation section, there are many drawbacks to using
+existing out of cluster k8s template tooling such as Helm and Kustomize to
+template VMs. Most notably, these solutions do not manage the underlying storage
+state, a critical requirement for templating stateful workloads like VMs.
+
+### Extend the containerdisks to include a VirtualMachine definition
+
+This could be an alternative to using a standardized format such as OVA/OVF to
+import and export VM templates. However, this would be a proprietary solution
+that is only supported by KubeVirt and does not allow interoperability with the
+larger virtualization ecosystem.
+
+### Extend instancetype.kubevirt.io to include networks and volumes
+
+As detailed in the Motivation, extending instance types would contradict their
+design goal of being generic and cluster-agnostic.
+
+## Scalability
+
+Adding this feature to KubeVirt should not affect its scalability. The new
+controller will only be concerned with the CRDs introduced by the design. The
+feature itself will mostly make use of the APIs already provided by KubeVirt.
+
+## Update/Rollback Compatibility
+
+The way of implementing this feature suggested by this design only requires
+minimal changes to the core KubeVirt repository. Therefore, the worst case when
+rolling back to an earlier version of KubeVirt would be that the feature is no
+longer available. However, all custom resources created from CRDs proposed in
+this design need to be removed before a rollback is possible.
```

Comment:
If deployment of the CRDs is handled by a separate operator I don't think this is true, rolling back kubevirt itself wouldn't do anything and an admin would need to manually remove the operator etc.

If we do go with a separate operator we should call out that an admin would need to disable the FG in KubeVirt to uninstall things correctly before rolling KubeVirt back.

Reply:
Adjusted the sentence a little, does it sound better now?